File: app/src/main/java/com/example/adam/myapplication/data/db/Converters.java
package com.example.adam.myapplication.data.db;

import android.arch.persistence.room.TypeConverter;

import java.util.Date;

public class Converters {

    @TypeConverter
    public Date dateFromLong(Long timestamp) {
        return new Date(timestamp);
    }

    @TypeConverter
    public Long longFromDate(Date date) {
        return date.getTime();
    }
}

File: app/src/main/java/com/example/adam/myapplication/data/db/task/Converters.kt
package com.example.adam.myapplication.data.db.task

import android.arch.persistence.room.TypeConverter

import java.util.Date

class Converters {

    @TypeConverter
    fun dateFromLong(timestamp: Long?): Date {
        return Date(timestamp!!)
    }

    @TypeConverter
    fun longFromDate(date: Date): Long? {
        return date.time
    }
}